### PR TITLE
NEW: Simple Multi-grid reconstruction method

### PR DESCRIPTION
**Pull request by carterbox**
## Purpose

Add a user-friendly wrapper around ptycho.reconstruct that implements a simple multi-grid reconstruction.

## Approach

Downsamples the ptychography problem by powers of 2 a requested number of times. Then performs the requested number of iterations at all levels. Each level provides the initial guess for the next higher-resolution level via upsampling. Parameters are upsampled using scipy.ndimage.zoom which uses spline interpolation. Used zoom because it was convenient, but also the splines provide smooth transition from low resolution to high.

Comparison of different re-gridding methods here:
https://anl.box.com/s/uwbh80n0vz37dmtjbbnsig53m53as0fr

### Submitter
- [x] Write a helpfully descriptive pull request title.
- [x] Organize changes into logically grouped commits with descriptive commit messages.
- [x] Document all new functions.
- [x] Click 'details' on the readthedocs check to view the updated docs.
- [x] Write tests for new functions or explain why they are not needed.
- [x] Address any complaints from pep8speaks.

### Reviewer
- [x] Actually read all of the code.
- [x] Run the new code yourself; the included tests should make this easy.
- [x] Write a summary of the changes as you understand them.
- [x] Thank the submitter.

### CODE REVIEW

**Review by stevehenke — APPROVED** [review]
These code changes add a basic multi-grid reconstruction capability to tike. The implementation is clean and tests are passing on my workstation. Thank you for your contribution!

File: src/tike/ptycho/solvers/options.py [hunk 1]
```diff
@@ -159,3 +161,31 @@ def __post_init__(self):
             self.psi,
             self.probe.shape,
         )
+
+    def resample(self, factor: float):
+        """Return a new PtychoParameter with the parameters scaled."""
+
+        return PtychoParameters(
+            probe=_resize_probe(self.probe, factor),
+            psi=_resize_probe(self.psi, factor),
```
[the remark this answers is not in the document]
Comment (carterbox) [reply src/tike/ptycho/solvers/options.py]: No, it was not. Now the resize functions are named better.

**Review by stevehenke — APPROVED** [review]
Changes look good. Automated tests are passing on my workstation.

**Comment by carterbox** [comment]
Rebased to remove an unrelated commit

**Pull request merged**